core/services/chainlink/config_general.go

```diff
@@ -1083,3 +1063,26 @@ var (
 	zeroURL        = url.URL{}
 	zeroSha256Hash = models.Sha256Hash{}
 )
+
+func (g *generalConfig) Logger() logger.Logger {
+	return g.lggr
+}
+
+func (g *generalConfig) NewLogger() logger.Logger {
+	lggrCfg := logger.Config{
+		LogLevel:       g.LogLevel(),
+		Dir:            g.LogFileDir(),
+		JsonConsole:    g.JSONConsole(),
+		UnixTS:         g.LogUnixTimestamps(),
+		FileMaxSizeMB:  int(g.LogFileMaxSize() / utils.MB),
+		FileMaxAgeDays: int(g.LogFileMaxAge()),
+		FileMaxBackups: int(g.LogFileMaxBackups()),
+	}
+	l, closeFn := lggrCfg.New()
```

Comment:
why isn't g.lggr being set but g.closeLogger is?

Reply:
🤦 No good reason. I thought it would make more sense to not have a function with a name like NewLogger() mutate anything so decided not to assign it here, but of course that makes no sense if we are assigning closeFn here.